src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -25,33 +17,39 @@
 
 @WebServlet("/login")
 public class LoginServlet extends HttpServlet {
+  private String clientId =
+      "12523748853-6h3fnh1kppht9s895ist2b5222ifh2u7.apps.googleusercontent.com"; // Client ID.
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    UserService userService = UserServiceFactory.getUserService();
-    boolean loggedIn = userService.isUserLoggedIn();
-    String urlToRedirect = "/search.html";
-
-    JSONObject userDetails = new JSONObject();
-    userDetails.put("loggedIn", loggedIn);
-    if (loggedIn) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      userDetails.put("userEmail", userEmail);
-      String logoutURL = userService.createLogoutURL(urlToRedirect);
-      userDetails.put("logoutURL", logoutURL);
-      // TODO: Implement OAuth for more robust email checking.
-      int start = userEmail.indexOf('@');
-      int end = userEmail.lastIndexOf('.');
-      if (start != -1 && end != -1) {
-        String schoolName = userEmail.substring(start + 1, end);
-        userDetails.put("schoolName", schoolName);
+  public LoginObject postHelper(HttpServletRequest request) throws IOException {
+    String idTokenString = request.getParameter("token");
+    UrlFetchTransport transport = UrlFetchTransport.getDefaultInstance();
+    GsonFactory json = GsonFactory.getDefaultInstance();
+    GoogleIdTokenVerifier verifier =
+        new GoogleIdTokenVerifier.Builder(transport, json)
+            .setAudience(Collections.singletonList(clientId))
+            .build();
+    try {
+      GoogleIdToken idToken = verifier.verify(idTokenString);
+      if (idToken != null) {
+        Payload payload = idToken.getPayload();
+        String userId = payload.getSubject();
+        return new LoginObject(userId, true);
       }
-
-    } else {
-      String loginURL = userService.createLoginURL(urlToRedirect);
-      userDetails.put("loginURL", loginURL);
+      return new LoginObject("NULL TOKEN", false);
+    } catch (GeneralSecurityException e) {
+      return new LoginObject("SECURITY ERROR", false);
+    } catch (IllegalArgumentException e) {
+      return new LoginObject("ARGUMENT ERROR", false);
     }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    LoginObject login = postHelper(request);
+    JSONObject js = new JSONObject();
```

Comment:
"js" is usually the abbreviation for Javascript, not JSON. Might want to change this to "json" or something like that